src/tlo/methods/equipment.py

```diff
@@ -0,0 +1,278 @@
+        dat = dat.groupby("Item_Code").transform(lambda x: x.fillna(x.mean()))
+
+        # Check no missing values
+        assert not dat.isnull().any()
+
+        return dat
+
+    def parse_items(self, items: Union[int, str, Iterable[int | str]]) -> Set[int]:
+        """Parse equipment items specified as an item_code (integer), an item descriptor (string), or an iterable of
+         either, and return as a set of item_code (integers). For any item_code/descriptor not recognised, a
+         `UserWarning` is issued."""
+
+        def check_item_codes_recognised(item_codes: set[int]):
+            if not item_codes.issubset(self._all_item_codes):
+                warnings.warn(f'Item code(s) "{item_codes}" not recognised.')
+
+        def check_item_descriptors_recognised(item_descriptors: set[str]):
+            if not item_descriptors.issubset(self._all_item_descriptors):
+                warnings.warn(f'Item descriptor(s) "{item_descriptors}" not recognised.')
+
+        # Make into a set if it is not one already
+        if isinstance(items, (str, int)):
+            items = set([items])
+        else:
+            items = set(items)
+
+        items_are_ints = all(isinstance(element, int) for element in items)
+
+        if items_are_ints:
+            check_item_codes_recognised(items)
+            # In the return, any unrecognised item_codes are silently ignored.
+            return items.intersection(self._all_item_codes)
+        else:
+            check_item_descriptors_recognised(items)  # Warn for any unrecognised descriptors
+            # In the return, any unrecognised descriptors are silently ignored.
+            return set(filter(lambda item: item is not None, map(self._item_code_lookup.get, items)))
+
+    def probability_all_equipment_available(
+        self, facility_id: int, item_codes: Set[int]
+    ) -> float:
+        """
+        Returns the probability that all the equipment item_codes are available
+        at the given facility.
+
+        It does so by looking at the probabilities of each equipment item being
+        available and multiplying these together to find the probability that _all_
+        are available.
+
+        NOTE: This will error if the facility ID or any of the item codes is not recognised.
+
+        :param facility_id: Facility at which to check for the equipment.
+        :param item_codes: Integer item codes corresponding to the equipment to check.
+        """
+
+        assert facility_id in self._all_fac_ids, f"Unrecognised facility ID: {facility_id=}"
+        assert item_codes.issubset(self._all_item_codes), f"At least one item code was unrecognised: {item_codes=}"
+
+        if self.availability == "all":
+            return 1.0
+        elif self.availability == "none":
+            return 0.0
+        return self._probabilities_of_items_available.loc[
+            (facility_id, list(item_codes))
```

Comment:
While above comment is still correct (I think) because of how[ indexing / subscripting of classes in Python is handled](https://stackoverflow.com/a/1685450) I now realise that `_probabilities_of_items_available` is a multi indexed series and so the tuple here is possibly meant to specifically hint that we're passing in a multi index key rather than index / column specifiers (as for a `DataFrame.loc`), so possibly keeping parenthesis makes sense!

Reply:
Indeed, that was the intention. In data frame indexing, one has to use the tuple for MultiIndex, otherwise it's ambiguous whether you want to refer to a column... so I did the same thing here for readability (for myself). My preference would be to keep the parantheses.

Reply:
Yep that makes sense!